espresso-server/app/src/androidTest/java/io/appium/espressoserver/lib/model/Element.kt

```diff
@@ -137,3 +143,39 @@ class Element(view: View) {
         }
     }
 }
+
+class ComposeElement(node: SemanticsNodeInteraction) : BaseElement {
+    @Suppress("JoinDeclarationAndAssignment")
+    @SerializedName(JSONWP_ELEMENT_KEY, alternate = [W3C_ELEMENT_KEY])
+    override val element: String
+
+    init {
+        element = UUID.randomUUID().toString()
+        ComposeViewCache.put(element, node)
+    }
+
+    companion object {
+        /**
+         * Retrieve cached view and return the SemanticsNodeInteraction
+         */
+        @Throws(AppiumException::class)
+        fun getNodeInteractionById(elementId: String?): SemanticsNodeInteraction? {
+            elementId ?: throw InvalidArgumentException("Cannot find 'null' element")
+            if (!ComposeViewCache.has(elementId)) {
+                throw NoSuchElementException(
```

Comment:
I assume this should throw stale element exception instead

Reply:
stale will be an element that previously was found and now became bad. In this case, the element never existed in the cache and maybe never found. This is in line with EspressoViewCache

Reply:
The cache is LRU one, which would remove older element if they expire. I would still prefer to throw stale element exception here and only throw not found in find element handler, and, probably, find active element one.

Reply:
changed